src/operations/query/filters/id.js

```diff
@@ -66,6 +66,84 @@ class FilterById extends BaseFilter {
         return query;
 
     }
+
+    /**
+     * Generates filter for parsedArgItem
+     * @return {import('mongodb').Filter<import('mongodb').DefaultSchema>[]}
+     */
+    filter() {
+        /**
+         * @type {import('mongodb').Filter<import('mongodb').DefaultSchema>[]}
+         */
+        const and_segments = [];
+
+        if (this.parsedArg.queryParameterValue.values) {
+            and_segments.push({
+                    $or: this.propertyObj.fields.flatMap((field) => {
+                            return {
+                                [this.parsedArg.queryParameterValue.operator]:
+                                this.filterByItems(field, this.parsedArg.queryParameterValue.values)
+                            };
+                        }
+                    ),
+                },
+            );
+        }
+
+        return and_segments;
+    }
+
+    /**
+     * Generate filter based of field and values
+     * @param {string} field
+     * @param {string[]} values
+     * @returns {Array<Object>}
+     */
+    filterByItems(field, values) {
+        if (this.enableGlobalIdSupport) {
```

Comment:
This flag should not be required anymore. There is a ticket to remove it. https://icanbwell.atlassian.net/browse/EFS-337